core/ipc/sink/common/src/main/java/org/opennms/core/ipc/sink/offheap/DispatchQueueServiceLoader.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.core.ipc.sink.offheap;
+
+import java.util.Optional;
+
+import org.opennms.core.ipc.sink.api.DispatchQueueFactory;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+public class DispatchQueueServiceLoader {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DispatchQueueServiceLoader.class);
+    private static BundleContext context;
+    private static volatile DispatchQueueFactory dispatchQueueFactory;
+
+    public BundleContext getBundleContext() {
+        return context;
+    }
+
+    public static void setBundleContext(BundleContext bundleContext) {
+        context = bundleContext;
+    }
+
+    public static Optional<DispatchQueueFactory> getDispatchQueueFactory() {
+        if (dispatchQueueFactory != null) {
+            return Optional.of(dispatchQueueFactory);
+        }
+
+        if (context != null) {
+            try {
+                dispatchQueueFactory = context.getService(context.getServiceReference(DispatchQueueFactory.class));
```

Comment:
I think you have to do null check on the reference before passing it to  `context.getService`.  In the previous case, it used to check this only when off-heap feature is enabled config is present.

Reply:
Yeah you are right. I'll add that too. I think it is still preferably to install the feature on sentinel additionally so we use the same impl.